op-acceptance-tests/tests/base/leadership_transfer_test.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+type conductorWithInfo struct {
+	stack.Conductor
+	info consensus.ServerInfo
+}
+
+// TestLeadershipTransfer checks if the leadership transfer works correctly on the conductors
+func TestLeadershipTransfer(gt *testing.T) {
+	t := devtest.SerialT(gt)
+	sys := presets.NewMinimal(t)
+	logger := testlog.Logger(t, log.LevelInfo).With("Test", "TestLeadershipTransfer")
+	tracer := t.Tracer()
+	ctx := t.Ctx()
+	logger.Info("Started L2 RPC connectivity test")
+
+	ctx, span := tracer.Start(ctx, "test chains")
+	defer span.End()
+
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	// Test all L2 chains in the system
+	for _, l2Chain := range sys.L2Networks() {
+		_, span = tracer.Start(ctx, "test chain")
+		defer span.End()
+
+		require.NotEmpty(t, l2Chain.Escape().Conductors(), "no conductors found in the L2 chain")
+
+		membership := dsl.NewConductor(l2Chain.Escape().Conductors()[0]).FetchClusterMembership()
+		if len(l2Chain.Escape().Conductors()) == 0 {
+			t.Skip("no conductor found in the input, skipping leadership transfer test")
+			continue
+		}
+		require.Equal(t, len(membership.Servers), len(l2Chain.Escape().Conductors()), "cluster membership does not match the number of conductors")
+
+		idToConductor := make(map[string]conductorWithInfo)
+		for _, conductor := range l2Chain.Escape().Conductors() {
+			idToConductor[string(conductor.ID())] = conductorWithInfo{conductor, consensus.ServerInfo{}}
+		}
+		for _, memberInfo := range membership.Servers {
+			conductor, ok := idToConductor[memberInfo.ID]
+			require.True(t, ok, "unknown conductor in cluster membership", memberInfo.ID)
+			conductor.info = memberInfo
+			idToConductor[memberInfo.ID] = conductor
+		}
+
+		leaderInfo, err := l2Chain.Escape().Conductors()[0].RpcAPI().LeaderWithID(ctx)
+		require.NoError(t, err, "failed to get current conductor info")
+
+		leaderConductor := idToConductor[leaderInfo.ID]
+
+		voters := []conductorWithInfo{leaderConductor}
+		for _, member := range membership.Servers {
+			if member.ID == leaderInfo.ID || member.Suffrage == consensus.Nonvoter {
+				continue
+			}
+
+			voters = append(voters, idToConductor[member.ID])
+		}
+
+		if len(voters) == 1 {
+			t.Skip("only one voter found in the cluster, skipping leadership transfer test")
+			continue
+		}
+
+		t.Run(fmt.Sprintf("L2_Chain_%s", l2Chain.String()), func(tt devtest.T) {
+			numOfLeadershipTransfers := len(voters)
+			for i := 0; i < numOfLeadershipTransfers; i++ {
+				oldLeader := voters[i]                 // no need for i%numOfLeadershipTransfers, since i is always less than numOfLeadershipTransfers
+				newLeader := voters[(i+1)%len(voters)] // same as i+1, except it becomes 0 when i+1 is numOfLeadershipTransfers
+
+				testTransferLeadershipAndCheck(t, voters, oldLeader, newLeader)
+			}
```

Comment:
nit: this is a matter of preference but, FWIW, declaring `numOfLeadershipTransfers` here was a bit confusing to me. Especially since you explicitly also use `len(voters)`. I'd say either just use `len(voters)` and perhaps add a comment as to what we're looping over, or just use `numOfLeadershipTransfers`.

Reply:
So the usage of modulo was to perform the loop over for the scenarios whenever i or i+1 got >= len(voters). That's because i's upper bound it numOfLeadershipTransfers and it can very well be (much) more than len(voters).

Though I do see your point, so I have added a more understandable comment and better var naming to convey the context better.